src/pcap/context.go

```diff
@@ -0,0 +1,54 @@
+package pcap
+
+// This file provides an early implementation of go1.20s context implementation.
+// It adds functionality to cancel a context with a cause and retrieve that cause
+// later. This is useful if you need concurrent programs to communicate the original
+// error that caused the execution to stop.
+// FIXME(maxmoehl): refactor to stdlib
+
+import (
+	"context"
+	"sync/atomic"
+)
+
+// &cancelCauseKey is the key at which the cause for cancellation is stored.
```

Comment:
That was not the intention. With your suggestion the sentence is no longer true, a variable of type int with value 0 is not the key, it's just the pointer to this very variable (also it's just copied from [here](https://cs.opensource.google/go/go/+/refs/tags/go1.20rc1:src/context/context.go;l=342-343)).

Reply:
Fair, then maybe we can add 2-3 more words about the address being used as guaranteed unique key? The `&` is easily overlooked at first glance. I noticed because you told me about the 'neat trick'.